app/scripts/metamask-controller.js

```diff
@@ -852,8 +855,83 @@ export default class MetamaskController extends EventEmitter {
             saveState: async () => {
               await this.keyringController.persistAllKeyrings();
             },
-            removeAccount: async (address) => {
-              await this.removeAccount(address);
+            addAccount: async (_address, origin, handleUserInput) => {
+              const { id: addAccountApprovalId } =
+                this.approvalController.startFlow();
+
+              const confirmationResult =
+                await this.approvalController.addAndShowApprovalRequest({
+                  origin,
+                  type: SNAP_MANAGE_ACCOUNTS_CONFIRMATION_TYPES.confirmAccountCreation,
+                  expectsResult: true,
```

Comment:
This property is only needed if we want the code that accepts the approval request to wait for a result from this code here so we can indicate if the process being accepted was successful.

If so, there is a `resultCallbacks` property in the returned object that we need to call `success` or `error` on.